pymatgen/symmetry/analyzer.py

```diff
@@ -1697,3 +1700,543 @@ def __str__(self):
+        """
+        MAGSPACEGROUP2MAGPOINTGROUP = os.path.join(os.path.dirname(__file__), "mspg2mpg_map.txt")
+      
+        glob_og_no = ".".join([str(x) for x in self._MagneticSpacegroup._data["og_number"]])
+
+        with open(MAGSPACEGROUP2MAGPOINTGROUP, 'r') as f:
+            for line in f:
+                og_no, og_label, pointgroup_no, pointgroup_label = line.split()
+                if og_no == glob_og_no:
+                    return (pointgroup_label, pointgroup_no)
+
+    def get_magnetic_spacegroup(self):
+        """
+        Get the magnetic spacegroup.
+
+        Returns:
+            pymatgen.symmetry.maggroups.MagneticSpaceGroup 
+        """
+        return self._MagneticSpacegroup
+
+    def get_domains(self):
+        """
+        Get all possible magnetic domains as a dictionary.
+
+        A magnetic domain is obtained when a symmetry operation of the crystallographic 
+        spacegroup is applied to a magnetic structure with either time reversal odd or 
+        even and the resulting magnetic structure is not covering. 
+
+        The keys() are the applied magnetic symmetry operations. 
+        The values() are the resulting magnetic structure.
+
+        Returns:
+            dict(pymatgen.core.operations.MagSymmOp : pymatgen.core.structure.Structure) 
+        """
+      
+        if not self._domains:
+            self._domains = self._get_domains()
+      
+        return self._domains
+
+    def get_symmetry_ops(self):
+        """
+        Get magnetic spacegroup operations for the magnetic structure.
+
+        Returns:
+            list(pymatgen.core.operations.MagSymmOp)
+        """ 
+        return self._MagneticSpacegroup.symmetry_ops
+
+    def get_bns(self):
+        """
+        Get the magnetic spacegroup in the BNS notation [1], e.g. (P1, 1.1). 
+
+        [1] Belov et al. (1957). Sov. Phys. Crystallogr. 2, 311-322 
+      
+        Returns:
+            (str, list(int)): Magnetic space group in a tuple of (symbol, number) for structure.
+        """
+        bns_label = self._MagneticSpacegroup._data["bns_label"]
+        bns_number = ".".join([str(x) for x in self._MagneticSpacegroup._data["bns_number"]])
+        return (bns_label, bns_number)  
+
+    def get_og(self):
+        """
+        Get the magnetic spacegroup in the OG notation [1], e.g. (P1, 1.1.1). 
+
+        [1] Opechowski and Guccione (1965). Magnetism, edited by G. T. Rado and H. Suhl, Vol. II, 
+            Part A, pp. 105-165. New York: Academic Press
+      
+        Returns:
+            (str, list(int)): Magnetic space group in a tuple of (symbol, number) for structure.
+        """
+        og_label = self._MagneticSpacegroup._data["og_label"]
+        og_number = ".".join([str(x) for x in self._MagneticSpacegroup._data["og_number"]])
+        return (og_label, og_number)  
+
+    def _same_MagneticStructure(self, struc1, struc2, pos_tolerance=0.0015, magmom_tolerance=0.3):
```

Comment:
Perhaps worth discussing -- I think this functionality might already be partially present in `StructureMatcher` or perhaps `CollinearMagneticStructureAnalyzer.match()`, or maybe this method has a more narrow application than I understand? Where can `struc1` and `struc2` come from here?